### ISSUE TITLE
no-unused-expressions allow for undefined function check and call

### ISSUE THREAD

**Issue by kentcdodds** [eslint/eslint#2864]
Whether or not this is good practice is not my request. I would like to be able to configure `no-unused-expressions` to allow me to do this:

```javascript
foo && foo();
```

without having to turn the rule completely off. As far as I can tell, this is currently not possible. Is there an option I can provide that allows me to do this? Is this something that could be considered?

**Comment by michaelficarra** [comment]
Did you try searching for duplicate issues first?

**Comment by kentcdodds** [comment]
Sorry, I did not. My bad... Let me do that. 😅

**Comment by kentcdodds** [comment]
I suppose it could be similar to [this](https://github.com/eslint/eslint/issues/2181)

**Comment by kentcdodds** [comment]
... and [this](https://github.com/eslint/eslint/issues/2418).

It appears that there's a fundamental misunderstanding of this rule on my part (and apparently, I'm not alone). I'll close this one. Thanks.

**Comment by michaelficarra** [comment]
And #2684.

**Comment by kentcdodds** [comment]
lol, I thought that `foo && foo()` would be too general so I didn't try searching for that.
